egs2/chime8_task1/asr1/local/da_wer_scoring.py

```diff
@@ -0,0 +1,690 @@
+            c_sess_stats = {
+                "session id": session,
+                "scenario": scenario_tag,
+                "num spk hyp": len(hypothesis.labels()),
+                "num spk ref": len(reference.labels()),
+                "tot utterances hyp": len(hypothesis),
+                "tot utterances ref": len(reference),
+            }
+
+            c_sess_stats.update({k: v for k, v in der_score.items()})
+            c_sess_stats.update({k: v for k, v in jer_score.items()})
+            asr_err_sess, asr_err_spk = compute_asr_errors(
+                sess_dir, hyp_segs, ref_segs, mapping, uem=c_uem
+            )
+
+        else:
+            if not len(hyp_segs) == len(ref_segs):
+                warnings.warn(
+                    "If oracle diarization was used, "
+                    "I expect the hypothesis to have the same number "
+                    "of utterances as the "
+                    "reference. Have you discarded some utterances "
+                    "(e.g. too long) ? "
+                    "These will be counted as deletions so be careful !"
+                )
+            asr_err_sess, asr_err_spk = compute_asr_errors(
+                sess_dir, hyp_segs, ref_segs, mapping=None, uem=c_uem
+            )
+            c_sess_stats = {
+                "session id": session,
+                "scenario": scenario_tag,
+                "num spk hyp": len(set([x["speaker"] for x in hyp_segs])),
+                "num spk ref": len(set([x["speaker"] for x in ref_segs])),
+                "tot utterances hyp": len(hyp_segs),
+                "tot utterances ref": len(ref_segs),
+            }
+            # add to each speaker, session id and scenario
+        [
+            x.update({"session_id": session, "scenario": scenario_tag})
+            for x in asr_err_spk
+        ]
+        c_sess_stats.update(asr_err_sess)
+        all_sess_stats.append(c_sess_stats)
+        all_spk_stats.extend(asr_err_spk)
+
+    sess_df = pd.DataFrame(all_sess_stats)
+    # pretty print because it may be useful
+    print(tabulate(sess_df, headers="keys", tablefmt="psql"))
+    # accumulate for all scenario
+    scenario_wise_df = sess_df.sum(0).to_frame().transpose()
+    scenario_wise_df["scenario"] = scenario_tag
+    # need to recompute these
+    scenario_wer = compute_wer(sess_df)
+    scenario_wise_df["wer"] = scenario_wer
+    if use_diarization:
+        scenario_der = compute_der(sess_df)
+        scenario_wise_df["diarization error rate"] = scenario_der
+        scenario_jer = compute_jer(sess_df)
+        scenario_wise_df["Jaccard error rate"] = scenario_jer
+    del scenario_wise_df["session id"]  # delete session
+
+    return scenario_wise_df, all_sess_stats, all_spk_stats
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        "This script is used for scoring according to the procedure outlined in"
+        " CHiME-7 DASR challenge website"
```

Comment:
It's in chime-8 folder and the website points to latest challenge. But it should be fine.

Reply:
It is okay this is not used anymore for scoring

Reply:
but people may want still to use it